src/Akeneo/Channel/Bundle/Query/Sql/SqlIsCategoryTreeLinkedToChannel.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Channel\Bundle\Query\Sql;
+
+use Akeneo\Channel\Component\Query\PublicApi\IsCategoryTreeLinkedToChannel;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class SqlIsCategoryTreeLinkedToChannel implements IsCategoryTreeLinkedToChannel
+{
+    private Connection $connection;
+
+    public function __construct($connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function byCategoryTreeId(int $categoryTreeId): bool
+    {
+        $sql = <<<SQL
+        SELECT EXISTS (
+            SELECT *
```

Comment:
```suggestion
            SELECT id
```

Doing a `SELECT *` is not optimized, as we could have a very long text column and so on. And we don't need that here.